s4e-web/cypress/integration/settings/settings-institution-layer.spec.ts

```diff
@@ -0,0 +1,76 @@
+/// <reference types="Cypress" />
+
+import { Login } from '../../page-objects/auth/auth-login.po';
+import { UserOptionsGoToSettings } from "../../page-objects/user-options/user-options-go-to-settings-profile.po";
+import { SettingsInstitutionLayer } from "../../page-objects/settings/settings-institution-layer.po";
+import { SettingsInstitutions } from "../../page-objects/settings/settings-institution-profile.po";
+import { MapLayers } from "../../page-objects/map/map-layers.po";
+import { ConfirmModal } from "../../page-objects/modal/confirm-modal.po";
+import { SettingsNav } from "../../page-objects/settings/settings-navigation.po"
+
+before(() => {
+  cy.fixture('users/zkAdmin.json').as('zkAdmin');
+  cy.fixture('layer-capability.xml').as('geoserverResponse');
+});
+
+describe('Institution Layer', () => {
+
+  beforeEach(function () {
+    cy.visit('/login')
+    Login
+      .loginAs(this.zkAdmin)
+    MapLayers.pageObject.getLayerCount();
```

Comment:
Why is such call necessary in the beforeEach?

Reply:
I check the number of layers in the application before the test. In the test, I add 1 layer to the institution and check if the number of layers has increased by 1

Reply:
Logging-in is understood. Don't the layer count assertions from `map-layers.po.ts` call pageObject themselves? E.g. in the function you use to verify the count has risen by 1, it reads:
```ts
  static sidebarLayersCountShouldBe(count: number) {
    MapLayers
      .pageObject
      .getSidebarLayers() // there the pageObject is called
      .should('have.length', count);

    return MapLayers;
  }
```
I understand the call `.as('layerCount')` in `getLayerCount` assigns the value under this to placeholder (?) `layerCount`, but then there are no references to it throughout the code (it would read `@layerCount`, at least according to docs I have loaded).
![image](https://user-images.githubusercontent.com/583197/105961939-75e53200-607f-11eb-8ab2-88c56b24bd6d.png)

Reply:
.should('have.length', count) it is only assertion, this function doesnt allow use the count because doesnt return the number.

Reply:
> .should('have.length', count) it is only assertion, this function doesnt allow use the count because doesnt return the number.

Okay, but it still doesn't explain why the call in the beforeEach is necessary. It seems to me that the line `    MapLayers.pageObject.getLayerCount();` can be safely removed, it's a no-op.

Reply:
I'd suggest you to try removing it, see if the tests pass:
- if they pass, leave it removed
- if they fail, re-add it and give me a heads-up that it is necessary after all

Reply:
I get every element for test in page object files but tests execute to the another file spec.ts. So I cant use directly cy.get(@layerCount) in spec.ts. I have to create another function in Page Object file for cy.get(@layerCount) and import and trigger to spec.js file where I have tests. I will think which solution will be better.

Reply:
it is necessary, without this test is failed 
![image](https://user-images.githubusercontent.com/59872096/105972895-67e9de00-608c-11eb-8473-674ce77d87db.png)

Reply:
Okay, I understand now. Sorry for the bother.

The pageObject.getLayerCount actually **sets** (and not gets) the observed layer count and puts it in `this.layerCount` (I didn't know you can access it this way) that you use for assertion later on. I'd rename it from `getLayerCount` to `setAliasLayerCount` for better clarity too.